src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
 public class WorkWithFile {
+    private static final String SUPPLY_STR = "supply";
+    private static final String BUY_STR = "buy";
+    private static final String RESULT_STR = "result";
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        ArrayList<String> data = readFile(fromFileName);
+        String operations = createReport(data);
+        writeDataToFile(toFileName, operations);
+    }
+
+    private ArrayList<String> readFile(String fileName) {
+        File fileToRead = new File(fileName);
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fileToRead))) {
+            ArrayList<String> operationsList = new ArrayList<>();
+            String line;
+
+            while ((line = br.readLine()) != null) {
+                operationsList.add(line);
+            }
+
+            return operationsList;
+        } catch (IOException e) {
+            throw new RuntimeException("File not found: ", e);
+        }
+    }
+
+    private String createReport(ArrayList<String> operationsList) {
+
+        int supplyOperations = 0;
+        int buyOperations = 0;
+        StringBuilder builder = new StringBuilder();
+
+        for (String operation : operationsList) {
+            String[] partsOfOperation = operation.split(",");
+            String type = partsOfOperation[0];
+            int quantity = Integer.parseInt(partsOfOperation[1]);
+
+            if (type.equals("buy")) {
+                buyOperations += quantity;
+            } else {
+                supplyOperations += quantity;
+            }
+        }
+
+        builder.append(SUPPLY_STR)
+                .append(COMMA)
+                .append(supplyOperations)
+                .append(LINE_SEPARATOR);
```

Comment:
```suggestion
                .append(System.lineSeparator());
```

for other places also